CNV12285: Adding critical CNV alerts to downstream documentation

mgarrellRH opened the pull request:
This PR addresses JIRA epic [CNV-12285](https://issues.redhat.com/browse/CNV-12285) ( https://issues.redhat.com/browse/CNV-12285 ).

This epic comprises 5 stories:
[CNV-3264](https://issues.redhat.com/browse/CNV-3264) ( https://issues.redhat.com/browse/CNV-3264 )
[CNV-13557](https://issues.redhat.com/browse/CNV-13557) ( https://issues.redhat.com/browse/CNV-13557 )
[CNV-13721](https://issues.redhat.com/browse/CNV-13721) ( https://issues.redhat.com/browse/CNV-13721 )
[CNV-13722](https://issues.redhat.com/browse/CNV-13722) ( https://issues.redhat.com/browse/CNV-13722 )
[CNV-13723](https://issues.redhat.com/browse/CNV-13723) ( https://issues.redhat.com/browse/CNV-13723 )

The epic asks to document critical CNV alerts. However, [CNV-13557](https://issues.redhat.com/browse/CNV-13557) and [CNV-13723](https://issues.redhat.com/browse/CNV-13723) do not have any Critical alerts. So they are outside the scope of this PR.

CP: 4.10 (Tech Preview only)

Preview: https://deploy-preview-40669--osdocs.netlify.app/openshift-enterprise/latest/virt/logging_events_monitoring/virt-virtualization-alerts.html

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
++
+[source,terminal]
+----
+export CDI_NAMESPACE="$(kubectl get deployment -A | grep cdi-operator | awk '{print $1}')"
```
In downstream documentation I believe we can drop this query and replace the CDI_NAMESPACE with openshift-cnv.
@akalenyu  do you agree?
If you want to keep the documentation aligned for some reason then we should replace "kubectly" with "oc".
I see the "kubectly"  still appears it other files as well.

mgarrellRH replied:
CDI_NAMESPACE instances have now been changed to openshift-cnv and "kubectl" replaced with "oc".

If we're going to drop the query, do we remove that step entirely? Or just change the CDI_NAMESPACE and kubectl references? If we're removing it entirely, do I need to remove the same query in the couple other places it appears?

sradco commented:
Hi @akalenyu @iholder-redhat @borod108 @andreyod @nunnatsa @yuhaohaoyu.
Can you please help with reviewing the runbooks, that are related to you, that @mgarrellRH  has created for downstream docs?

sradco commented on modules/virt-cnv-network-alerts.adoc:
```diff
@@ -0,0 +1,183 @@
++
+[source,terminal]
+----
+export NAMESPACE="$(oc get deployment -A | grep cluster-network-addons-operator | awk '{print $1}')"
```
@oshoval I believe we can drop this command lines 23-27 and replace $NAMESPACE with openshift-cnv. Do you agree?

oshoval replied:
We can, but what about the following:

export NAMESPACE=openshift-cnv
The default downstream namespace is `openshift-cnv`.
It can be verified by the command ...

This will support potential cases where the namespace is changed.
Also this whole file looks like a command tips, which are connected to each other

sradco replied:
@oshoval  Yes, do we allow to deploy CNV on a different namespace?

sradco replied:
@mgarrellRH Lets keep this as is for safety.

oshoval replied:
> @oshoval Yes, do we allow to deploy CNV on a different namespace?

Yes, the code allows arbitrary namespace for the CNAO operator
(for example the U/S one is different from the D/S one)

oshoval replied:
btw you can use
`oc get deployment -A --field-selector metadata.name=cluster-network-addons-operator  -o=custom-columns=NS:.metadata.namespace --no-headers`

sradco replied:
@oshoval I like this change. It is a more elegant. I see that we use the awk in other runbooks and I think we should create a PR to update all . 
I think for now we can keep as is and Ill open an issue for the runbooks to fix it.
Once its fixed u/s we will fix it also d/s.

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
++
+[source,terminal]
+----
+oc logs -n $openshift-cnv deployment/cdi-deployment
```
@mgarrellRH , Lets revert this and keep the $NAMESPACE instead of $openshift-cnv

apinnick replied:
@sradco At this late date, I recommend that you stay with the default "openshift-cnv" instead of a "$NAMESPACE" variable. It is safer, as you say above. :-)

apinnick replied:
Based on chat with Shirly, keeping $NAMESPACE.

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
++
+[source,terminal]
+----
+export openshift-cnv="$(oc get deployment -A | grep cdi-operator | awk '{print $1}')"
```
@mgarrellRH , Lets revert this and keep the NAMESPACE instead of openshift-cnv

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
++
+[source,terminal]
+----
+export openshift-cnv="$(oc get deployment -A | grep cdi-operator | awk '{print $1}')"
```
@mgarrellRH , Lets revert this and keep the NAMESPACE instead of openshift-cnv

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
++
+[source,terminal]
+----
+oc -n $openshift-cnv get deploy -l cdi.kubevirt.io
```
@mgarrellRH , Lets revert this and keep the $NAMESPACE instead of $openshift-cnv

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
++
+[source,terminal]
+----
+oc -n $openshift-cnv describe pods <corresponding_pod_name>
```
@mgarrellRH , Lets revert this and keep the $NAMESPACE instead of $openshift-cnv

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
+----
+oc -n $openshift-cnv describe pods <corresponding_pod_name>
+
+oc -n $openshift-cnv logs <corresponding_pod_name>
```
@mgarrellRH , Lets revert this and keep the $NAMESPACE instead of $openshift-cnv

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
++
+[source,terminal]
+----
+oc -n $openshift-cnv get pods -l name=cdi-operator
```
@mgarrellRH , Lets revert this and keep the $NAMESPACE instead of $openshift-cnv

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
++
+[source,terminal]
+----
+export openshift-cnv="$(oc get deployment -A | grep cdi-operator | awk '{print $1}')"
```
@mgarrellRH , Lets revert this and keep the NAMESPACE instead of openshift-cnv

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
++
+[source,terminal]
+----
+oc -n $openshift-cnv describe pods -l name=cdi-operator
```
@mgarrellRH , Lets revert this and keep the $NAMESPACE instead of $openshift-cnv

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
+----
+oc -n $openshift-cnv describe pods -l name=cdi-operator
+
+oc -n $openshift-cnv logs -l name=cdi-operator
```
@mgarrellRH , Lets revert this and keep the $NAMESPACE instead of $openshift-cnv

sradco commented on modules/virt-cnv-network-alerts.adoc:
```diff
@@ -0,0 +1,183 @@
++
+[source,terminal]
+----
+oc -n openshift-cnv describe pods -l name=cluster-network-addons-operator
```
@mgarrellRH please also update here openshift-cnv to $NAMESPACE

sradco replied:
@mgarrellRH Hi, You missed updating openshift-cnv to $NAMESPACE here

apinnick commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+
+.Impact
+
+If this alert consistently oxxurs, the node’s CPU resources are not being optimally used and could be overloaded.
```
typo: oxxurs

sradco commented on modules/virt-cnv-hco-alerts.adoc:
```diff
@@ -0,0 +1,73 @@
+
+.Meaning
+
+The Hyperconverged Cluster Operator (HCO) configures kubevirt and its supporting operators and overwrites its operands if unexpected changes occur.
```
@nunnatsa Should we replace KubeVirt here with OpenShift Logging?

sradco commented:
@mgarrellRH we are mentioning KubeVirt in many places. We we are referring to the product I think we should replace with OpenShift Virtualization.

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+
+.Impact
+
+Virt-operator has likely lost the connection to the apiserver. Cluster-level actions such as upgrading and controller reconciliation likely do not function. Customer workloads such VMs and VMIs should not be affected.
```
Customer workloads such VMs -> Customer workloads such as VMs

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+
+.Mitigation
+
+There can be for several reasons for a low virt-operator count. Identify the root cause and take appropriate action.
```
There can be for several  -> There can be several

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+
+.Mitigation
+
+There can be for several reasons for virt-operator pods being down. Identify the root cause and take appropriate action.
```
There can be for several -> There can be several

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+* Monitoring the life-cycle of top-level controllers such as virt-controller, virt-handler, and virt-launcher
+* Managing the reconciliation of top-level controllers
+
+In addition, virt-operator is responsible for cluster-wide tasks such as certificate rotation and some infrastructure management.
```
In addition, virt-operator - > In addition, the virt-operator

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+* Monitoring the life-cycle of top-level controllers such as virt-controller, virt-handler, and virt-launcher
+* Managing the reconciliation of top-level controllers
+
+In addition, virt-operator is responsible for cluster-wide tasks such as certificate rotation and some infrastructure management.
```
In addition, virt-operator - > In addition, the virt-operator

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+* Monitoring the life-cycle of top-level controllers such as virt-controller, virt-handler, and virt-launcher
+* Managing the reconciliation of top-level controllers
+
+In addition, virt-operator is responsible for cluster-wide tasks such as certificate rotation and some infrastructure management.
```
In addition, virt-operator - > In addition, the virt-operator

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+* Monitoring the life-cycle of top-level controllers such as virt-controller, virt-handler, and virt-launcher
+* Managing the reconciliation of top-level controllers
+
+In addition, virt-operator is responsible for cluster-wide tasks such as certificate rotation and some infrastructure management.
```
In addition, virt-operator - > In addition, the virt-operator

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+oc get pods -n $NAMESPACE -l=kubevirt.io=virt-handler
+----
+
+* Determine the name of worker node the pod is deployed on for each virt-handler pod.
```
of worker -> of the worker

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+
+.Impact
+
+This alert does not mean that the failure of rollouts of all virt-handler DaemonSet. It should not affect the normal VM life-cycle if the cluster is not overloaded.
```
@iholder-redhat What does this sentence mean?

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+
+.Meaning
+
+Some virt-handler DaemonSet fail to roll out after 15 minutes. This alert suggests that, in the cluster, at least one worker node does not have the virt-handler DaemonSet pod successfully rolled out in the given time.
```
fail -> fails

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+oc get daemonset virt-handler --all-namespaces -o yaml | jq .status
+----
+
+Check the health of the clusters nodes.
```
clusters -> cluster

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+
+A VMI object is always associated with a pod during it’s lifetime. However, the pod instance might change over time because of VMI migration.
+
+This alert occur when no ready virt-controllers are detected for the last five minutes.
```
occur -< occurs

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+
+The virt-controller is responsible for monitoring virtual machine instances (VMIs) and managing the associated pods by creating and managing the life-cycle of the pods associated with the VMI objects.
+
+A VMI object is always associated with a pod during it’s lifetime. However, the pod instance might change over time because of VMI migration.
```
it's -> its

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+
+Virt-controller is responsible for monitoring virtual machine instances (VMIs) and managing the associated pods by creating and managing the life-cycle of the pods associated with the VMI objects.
+
+A VMI object is always associated with a pod during it’s lifetime. However, the pod instance might change over time because of VMI migration.
```
it's -> its

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+
+.Impact
+
+If some virt-controllers are running but not ready in the past five minutes, then virt-controller becomes a single point of failure.
```
then virt-controller -> then the virt-controller

sradco commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,1347 @@
+
+.Mitigation
+
+Consider changing container memory limit.
```
changing container -> changing the container

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
+
+.Mitigation
+
+A common issue when opting in to golden images auto-polling is not having a default storage class set.
```
in to -> into

sradco commented on modules/virt-cnv-storage-alerts.adoc:
```diff
@@ -0,0 +1,377 @@
+
+Such PVCs always get updated with the latest version, serving as a trustworthy clone source for VM created from the latest image of an OS.
+
+This alert occurs when a DataImportCron fails to create a corresponding PVC or keep its corresponding PVC updated because it already has `latest` version.
```
has the `latest`

sradco commented on virt/logging_events_monitoring/virt-virtualization-alerts.adoc:
```diff
@@ -0,0 +1,16 @@
+:FeatureName: OpenShift Virtualization alerts
+include::snippets/technology-preview.adoc[]
+
+OpenShift Virtualization uses alerts to provide additional information for its metrics. Each alert has a corresponding Operator.
```
Each alert has a corresponding alert runbook.

mgarrellRH replied:
Would we tell downstream customers that each alert has an upstream runbook?

mgarrellRH commented:
> The sentence "Each alert has a corresponding Operator" has no meaning as is.. We want to say that we provide runbooks for helping diagnose and solve issues

The first sentence does say "OpenShift Virtualization uses alerts to provide additional information for its metrics." But I've changed the sentence to say "OpenShift Virtualization uses alerts to troubleshoot and resolve problems, as well as to provide additional information for its metrics."

The "corresponding Operator" sentence means that the alerts are organized based on each Operator. I've rewritten the sentence as "Each alert corresponds with the Virt, Network, or SSP Operator, which you use to manage virtual machines."

Does that help?

sradco commented:
> > The sentence "Each alert has a corresponding Operator" has no meaning as is.. We want to say that we provide runbooks for helping diagnose and solve issues
> 
> The first sentence does say "OpenShift Virtualization uses alerts to provide additional information for its metrics." But I've changed the sentence to say "OpenShift Virtualization uses alerts to troubleshoot and resolve problems, as well as to provide additional information for its metrics."
> 
> The "corresponding Operator" sentence means that the alerts are organized based on each Operator. I've rewritten the sentence as "Each alert corresponds with the Virt, Network, or SSP Operator, which you use to manage virtual machines."
> 
> Does that help?

Next version also VM alerts which are not operator alerts. I also don't think we should say they are "operator" alerts. They are alerts for OpenShift Virtualization. Why does the users need to know the sub-operators?

OpenShift Virtualization provides alerts and each alerts has a corresponding runbook to better explain the alert, its impact, how to diagnose it and how to mitigate it.

sradco commented:
/lgtm

yossisegev commented on modules/virt-cnv-network-alerts.adoc:
```diff
@@ -0,0 +1,44 @@
+export KMP_NAME="$(oc get pod -A --no-headers -l control-plane=mac-controller-manager | awk '{print $2}')"
+----
+
+* Check the Kubemacpool-manager pod's pod description and logs.
```
I think "Kubemacpool-manager pod description" might sound better.

RamLavi replied:
+1

yossisegev commented on modules/virt-cnv-network-alerts.adoc:
```diff
@@ -0,0 +1,44 @@
+
+oc logs -n $KMP_NAMESPACE $KMP_NAME
+----
+
```
And - look for what? An entry of `Alerts`? 
An output example might be useful.

sradco replied:
@RamLavi Can you please help here?

RamLavi replied:
>@RamLavi Would it help to add something in the Reason section saying "The KubeMacPool-manager pod may be down for several reasons, such as" and listing a few of the possible reasons?

That's a strong no on that . 
Only reason for this alert to go on is if there's a bug in kubemacpool. kubemacpoolDown alert is relevant because of the severity of the outcome when down (meaning, no vms can be created/updated), but it isn't to say that it is more prone to be down than any other pod.

>Or perhaps in this oc logs step, we could say "Check the Kubemacpool-manager pod description and logs to determine the source of the problem."?

That is better, I would vote for that. I would also address the sysAdmin to open a case about it, as it is probably not something they can fix.

dbasunag commented on modules/virt-cnv-network-alerts.adoc:
```diff
@@ -0,0 +1,44 @@
++
+[source,terminal]
+----
+export KMP_NAMESPACE="$(oc get pod -A --no-headers -l control-plane=mac-controller-manager | awk '{print $1}')"
```
Should this be `openshift-cnv`? Since this documentation is for downstream?

mgarrellRH replied:
@dbasunag During the code review, we debated about making it `openshift-cnv` for downstream. And at first, we did change it.

But then, it was agreed to keep it as a generic variable name in case future cases arise where being specific would require changing things. You can refer to earlier discussion in this PR about this issue.

dbasunag commented on modules/virt-cnv-ssp-alerts.adoc:
```diff
@@ -0,0 +1,132 @@
++
+[source,terminal]
+----
+export NAMESPACE="$(oc get deployment -A | grep ssp-operator | awk '{print $1}')"
```
Same comment about the namespace here. Should this be `openshift-cnv`?

mgarrellRH replied:
Same reply as previous. We initially did change it, but decided to keep it generic for future cases.

dbasunag commented on modules/virt-cnv-ssp-alerts.adoc:
```diff
@@ -0,0 +1,132 @@
+
+.Troubleshoot
+
+* Check the virt-template-validator’s Operator pod namespace:
```
Should this be `ssp-operator`?

mgarrellRH replied:
The upstream runbook source says `virt-template-validator`. But you may be correct.

@sradco Is the correct version `virt-template-validator` or `ssp-operator` in the SSPTemplateValidatorDown
 alert?

sradco replied:
@borod108 Can you please help here?

mgarrellRH replied:
@sradco Is there an answer for this question about `virt-template-validator` or `ssp-operator`?

sradco replied:
I don't know the answer to that.. @borod108 ?

dshchedr replied:
virt-template-validator - is a part of ssp-operator. For debugging this issue it would be useful to check both:

_ssp-operator pods_ and _virt-template-validator pods_

dbasunag commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+export NAMESPACE="$(oc get kubevirt -A -o custom-columns="":.metadata.namespace)"
+----
+
+. Run this command:
```
It would be nice to have some explanation on what is to expect after running the command or why it is being run, specially since we are checking the deployment again at line 112.

sradco replied:
@iholder-redhat Can you please help here?

dshchedr replied:
I think here the Trouleshooting paragraph is mixed with Resolution. 

**Troubleshouting (gathering info):**

1. Check deployment status for available replicas and conditions:
`$ oc -n $NAMESPACE get deployment virt-controller -o yaml`

2. check if virt-controller pods exists and their status:
`oc -n $NAMESPACE get pod | grep virt-controller`

3. check virt-controller pods events:
`oc -n $NAMESPACE describe pod <virt-controller pod> `

4. check virt-controller pods logs:
`oc -n $NAMESPACE logs <virt-controller pod> `

5. . Check if there are issues with the nodes, such as if the nodes are in a `NotReady` state.



**Resolution:** - something like we have for another alerts:

There are several reasons for no virt-controller pods being in a `Ready` state. Identify the root cause and take appropriate action.

Otherwise, open a support issue and provide the information gathered in the troubleshooting process.

dbasunag commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
++
+[source,terminal]
+----
+oc -n $NAMESPACE get deploy virt-controller -o yaml
```
Even though `deploy` would work fine, it would be good to be consistent and use either `deployment` or `deploy`. I feel, this should be updated in all the places.

mgarrellRH replied:
@dbasunag Does this comment also apply to the `oc -n $NAMESPACE describe deploy virt-controller` command that appears after `oc -n $NAMESPACE get deploy virt-controller -o yaml`?

dbasunag replied:
@mgarrellRH , Yes. There are multiple instances of `deploy` vs `deployment`. Both command will work, just for the sake of consistency, I am making this request.

mgarrellRH replied:
@dbasunag I suspected as much. I have changed all instances to `deployment`.

dbasunag commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+[source,terminal]
+----
+oc get nodes
+----
```
Like the `Resolution` sections for other alerts above, it would be good to clarify here what all information should be added to support case, in case it comes to that.

sradco replied:
@yuhaohaoyu Can you please help here?

mgarrellRH replied:
@yuhaohaoyu What information would be useful for a user to send in a support ticket?

In other places in these alerts, we've said "Open a support issue and provide the information gathered in the troubleshooting process."

Since this question/comment has come up several times, I'm hoping for some clarity on what is important to tell a user.

mgarrellRH replied:
Or @kbidarkar would you have any insight on what information would be useful for a user to send in a support ticket? and if saying "Open a support issue and provide the information gathered in the troubleshooting process" is sufficient?

dshchedr replied:
"Open a support issue and provide the information gathered in the troubleshooting process" - this sounds good for me. About troubleshooting - see my previous comment

dbasunag commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+export NAMESPACE="$(oc get kubevirt -A -o custom-columns="":.metadata.namespace)"
+----
+
+. Verify that there are any running virt-api pods.
```
Can this line be rephrased please? Possibly by replacing `any` with `no` or replacing `that` with `if`

dbasunag commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+----
+
+. Check if there are issues with the nodes, such as if the nodes are in a `NotReady` state.
++
```
It would be good to include here, what all information needs to be collected for support case, if it comes to that, just like other alert's resolution section.

dbasunag commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+oc -n $NAMESPACE get pods -l kubevirt.io=virt-api
+----
+
+. View the pods' logs using `oc logs` and pod status using `oc describe`.
```
`pod` should be replaced with `pods'` and possibly `status` with `statuses`, for consistency with `pods' logs`.

dbasunag commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+
+Virt-api pods can be down for several reasons. Identify the root cause and take appropriate action.
+
+. Check the status of the virt-api deployment to find out more information. These commands provide the associated events and show if there are any issues with pulling an image or a crashing pod.
```
It would be good to include, what all information should be collected in case a support ticket is deemed necessary.

dbasunag commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+.Resolution
+
+Check the virt-controller logs to determine if it cannot connect to the API server at all. If so, delete the pod to force a restart. The issue normally relates to DNS or CNI issues outside of the scope of this alert.
+
```
We might want to document, what all information should be collected if a support case is deemed necessary.

mgarrellRH replied:
@dshchedr Would you be able to provide some insight as to what information would be good to collect for a support ticket?

In other places in these alerts, we've said "Open a support issue and provide the information gathered in the troubleshooting process."

Is that enough or should more specific information be listed?

dshchedr replied:
In general, checking logs might be enough. But I would also add verification of this possible reasons: 

* Node resource exhaustion
* Not enough memory on the cluster

dbasunag commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+
+If the virt-handler cannot connect to the API server, delete the pod to force a restart. The issue normally relates to DNS or CNI issues outside of the scope of this alert. Identify the root cause and take appropriate action.
+
+//VirtOperatorDown Alert
```
Same comment as above.

dshchedr replied:
This **Troubleshooting** and **Resolution** for this alert should be the similar to alert: VirtControllerDown

just change virt-controller to virt-operator

dbasunag commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+
+.Resolution
+
+There are several reasons for virt-operator pods being down. Identify the root cause and take appropriate action.
```
Same comment as above here.

dshchedr commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+
+The virt-operator deployment has a default replica of two pods with one leader pod holding a leader lease, indicating an operating virt-operator pod.
+
+This alert indicates a failure at the cluster level cluster. Critical cluster-wide management functionalities such as certification rotation, upgrade, and reconciliation of controllers may be temporarily unavailable.
```
..  cluster level cluster ..
is this typo?

yuhaohaoyu reviewed (commented):
--> known reasons (listed as below) for the

yuhaohaoyu commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+.Resolution
+
+There are several reasons why detection of no running virt-controller occurs. Identify the root cause and take appropriate action.
+
```
"reasons why" --> "known reasons (listed as below) why the"

mgarrellRH replied:
@yuhaohaoyu Changed to "There are several known reasons why the detection of no running virt-controller occurs. Identify the root cause from the list of possible reasons and take appropriate action."

We try to avoid directional words like "above" or "below" in downstream docs. Does this change work?

yuhaohaoyu replied:
> @yuhaohaoyu Changed to "There are several known reasons why the detection of no running virt-controller occurs. Identify the root cause from the list of possible reasons and take appropriate action."
> 
> We try to avoid directional words like "above" or "below" in downstream docs. Does this change work?

Understood. @mgarrellRH Thanks for explanation. The change works for my view.

yuhaohaoyu commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+
+* The API server overloads, causing timeouts. Check the API server metrics and details, such as response times and overall calls.
+
+* The virt-operator pod cannot reach the API server. A common cause is a network connectivity problem, such as DNS issues on the node. Check the virt-operator logs to verify that the pod can connect to the API server at all.
```
"A common cause ..... on the node." --> "Common causes are network connectivity problems and DNS issues on the node."

mgarrellRH commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,487 @@
+
+[NOTE]
+====
+Do not use a name for the virt-controller pod that is similar to virt-controller-7888c64d66-dzc9p.
```
@dbasunag The upstream note said "virt-controller pod should have name similar to virt-controller-7888c64d66-dzc9p. There could be several pods that run virt-controller."

I've made a new edit to say "A virt-controller pod name example to virt-controller-7888c64d66-dzc9p. There may be several pods that run virt-controller."

Does the new edit clarify the note's meaning?

dbasunag replied:
@mgarrellRH yes, this edit is good. Thanks!

yossisegev commented on modules/virt-cnv-network-alerts.adoc:
```diff
@@ -0,0 +1,44 @@
+
+Network alerts provide information about problems for the CNV Network Operator.
+
+//KubeMacPoolDown Alert
```
Redundant leftover?

yossisegev replied:
> If you mean the //KubeMacPoolDown thing,
That's the one I was talking about. I just wasn't sure if it's inline documentation, or an outdated part that was commented-out instead of totally removed. Now I understand it's documentation.

yossisegev commented:
Network part - LGTM

geetikakay commented:
LGTM

geetikakay commented on modules/virt-cnv-ssp-alerts.adoc:
```diff
@@ -0,0 +1,132 @@
+----
+export NAMESPACE="$(oc get deployment -A | grep ssp-operator | awk '{print $1}')"
+
+oc -n $NAMESPACE get pods -l name=virt-template-validator
```
oc -n $NAMESPACE get pods -l name=virt-template-validator --> this command might not result in anything. 

$ oc -n $NAMESPACE get pods -l name=virt-template-validator
No resources found in openshift-cnv namespace.


$ oc -n $NAMESPACE get  pods -l kubevirt.io=virt-template-validator
NAME                                       READY   STATUS    RESTARTS   AGE
virt-template-validator-6685b5f546-6d2jp   1/1     Running   0          37m
virt-template-validator-6685b5f546-br25p   1/1     Running   0          37m

geetikakay replied:
I think this works in lastest version the way you mentioned. I was checking in bit older version.
Please resolve this comment.

$ oc -n $NAMESPACE get pods -l name=virt-template-validator
NAME                                       READY   STATUS        RESTARTS   AGE
virt-template-validator-58645467cf-2lpz6   1/1     Terminating   0          20d
virt-template-validator-58645467cf-4jmdb   1/1     Running       0          12d
virt-template-validator-58645467cf-krjpl   1/1     Running       0          20d

geetikakay commented on modules/virt-cnv-ssp-alerts.adoc:
```diff
@@ -0,0 +1,132 @@
+
+oc -n $NAMESPACE get pods -l name=virt-template-validator
+
+oc -n $NAMESPACE describe pods -l name=virt-template-validator
```
ditto

dshchedr commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,483 @@
+oc get nodes
+----
+
+.Resolution
```
I would move most of steps to troubleshooting section

dshchedr commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,483 @@
+oc -n $NAMESPACE get pods -l kubevirt.io=virt-api
+----
+
+.Resolution
```
I would move most of steps to troubleshooting section

dshchedr commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,483 @@
+
+. View the pods' logs using `oc logs` and the pods' statuses using `oc describe`.
+
+.Resolution
```
same: move all troubleshooting steps to Troubleshooting

dshchedr commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,483 @@
+oc -n $NAMESPACE describe deploy virt-api
+----
+
+. Check if there are issues with the nodes, such as if the nodes are in a `NotReady` state.
```
> if the nodes are in a `NotReady` state
 ... or overloaded

dshchedr commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,483 @@
+
+.Resolution
+
+Virt-api pods can be down for several reasons. Identify the root cause and take appropriate action.
```
in this alert we are talking about big amount of failed api calls, not about virt-api pods down.

I think need to re-phrase it to something like:

There are several reason for high rate of failed REST calls. Identify the root cause and take appropriate action.

dshchedr commented:
LGTM

dbasunag commented on modules/virt-cnv-ssp-alerts.adoc:
```diff
@@ -0,0 +1,134 @@
+----
+export NAMESPACE="$(oc get deployment -A | grep ssp-operator | awk '{print $1}')"
+
+export NAMESPACE="$(oc get deployment -A | grep virt-template-validator | awk '{print $1}')"
```
This export will override the last export on line 111.

dbasunag commented on modules/virt-cnv-ssp-alerts.adoc:
```diff
@@ -0,0 +1,134 @@
++
+[source,terminal]
+----
+export NAMESPACE="$(oc get deployment -A | grep ssp-operator | awk '{print $1}')"
```
Since this one is not being used below, may be we can have this removed?

dbasunag commented:
LGTM

lmandavi commented on modules/virt-cnv-network-alerts.adoc:
```diff
@@ -0,0 +1,44 @@
+
+.Description
+
+The KubeMacPool is responsible for allocating MAC Addresses and preventing MAC Address conflicts.
```
Recommend condensing the sentence: KubeMacPool allocates MAC addresses and prevents MAC address conflicts.

lmandavi commented on modules/virt-cnv-network-alerts.adoc:
```diff
@@ -0,0 +1,44 @@
+
+.Reason
+
+If the KubeMacPool-manager pod is down, then creating `VirtualMachine` objects fails.
```
Per ISG's subject-verb agreement guidance, the second part of the sentence is missing the subject for the verb "creating". Recommend: If the KubeMacPool-manager pod is down, the creation of 'VirtualMachine" objects fail.

lmandavi commented on modules/virt-cnv-network-alerts.adoc:
```diff
@@ -0,0 +1,44 @@
+//
+// * virt/logging_events_monitoring/virt-events.html/virt-virtualization-alerts.adoc
+
+[id="virt-cnv-network-alerts_{context}"]
```
Add the content type attribute as described in the Content type attribute requirements document. Here it would be:
:_content-type: PROCEDURE

lmandavi commented on modules/virt-cnv-ssp-alerts.adoc:
```diff
@@ -0,0 +1,138 @@
+
+.Description
+
+The SSP Operator’s pod is up, but the pod's reconcile cycle consistently fails. This failure includes failure to update the resources it is responsible for, failure to deploy the template validator, or failure to deploy or update the common templates.
```
suggest: failure to update the resources for which it is responsible...

lmandavi commented on modules/virt-cnv-network-alerts.adoc:
```diff
@@ -0,0 +1,44 @@
++
+[source,terminal]
+----
+oc describe pod -n $KMP_NAMESPACE $KMP_NAME
```
This applies to all lines that have the oc command syntax. The oc command should be preceded with $ and space:
$ oc describe pod...

lmandavi commented on modules/virt-cnv-ssp-alerts.adoc:
```diff
@@ -0,0 +1,138 @@
+[id="virt-cnv-ssp-alerts_{context}"]
+= SSP alerts
+
+SSP alerts provide information about problems for the CNV SSP Operator.
```
Instead of CNV SSP Operator, can this be OpenShift Virtualization SSP Operator?

sjhala-ccs commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,542 @@
+
+. Check the status of the virt-api deployment to find out more information. These commands provide the associated events and show if there are any issues with pulling an image or a crashing pod.
++
+[source,terminal]
```
split into separate code blocks

sjhala-ccs commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,542 @@
+$ oc get logs <virt-controller-pod>
+----
+
+An example of a virt-controller pod name in the logs is virt-controller-7888c64d66-dzc9p. However, there may be several pods that run virt-controller.
```
virt-controller-7888c64d66-dzc9p should be in backticks

sjhala-ccs commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,542 @@
+
+Check the virt-controller logs to determine if the virt-controller pod cannot connect to the API server at all. If so, delete the pod to force a restart.
+
+Additionally, verify if node resource exhaustion or not having enough memory on the cluster is causing the connection failure is node resource exhaustion
```
I think the `is node resource exhaustion` at the end of the sentence is a typo

sjhala-ccs commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,542 @@
+$ oc get logs <virt-operator-pod>
+----
+
+An example of a virt-operator pod name in the logs is virt-operator-7888c64d66-dzc9p. However, there may be several pods that run virt-operator.
```
virt-operator-7888c64d66-dzc9p in backticks

sjhala-ccs commented on modules/virt-cnv-virt-alerts.adoc:
```diff
@@ -0,0 +1,542 @@
+* The virt-operator pod cannot reach the API server. Common causes are network connectivity problems and DNS issues on the node. Check the virt-operator logs to verify that the pod can connect to the API server at all.
++
+[source,terminal]
+----
```
split code blocks

sjhala-ccs commented:
/cherrypick enterprise-4.10
